src/components/Pressable/GenericPressable/GenericPressable.js

```diff
@@ -0,0 +1,149 @@
+import React, {useCallback, useEffect, useMemo, forwardRef} from 'react';
+import {Pressable} from 'react-native';
+import _ from 'lodash';
+import Accessibility from '../../../libs/Accessibility';
+import HapticFeedback from '../../../libs/HapticFeedback';
+import KeyboardShortcut from '../../../libs/KeyboardShortcut';
+import styles from '../../../styles/styles';
+import genericPressablePropTypes from './PropTypes';
+
+const parseStyleFromFunction = (style, state) => (_.isFunction(style) ? style(state) : style);
+
+const getCursorStyle = (isDisabled, isText) => {
+    if (isDisabled) {
+        return styles.cursorDisabled;
+    }
+    if (isText) {
+        return styles.cursorText;
+    }
+    return styles.cursorPointer;
+};
+
+const GenericPressable = forwardRef((props, ref) => {
+    // eslint-disable-next-line react/destructuring-assignment
+    const {
+        children,
+        onPress,
+        onLongPress,
+        onKeyPress,
+        disabled,
+        style,
+        accessibilityHint,
+        shouldUseHapticsOnLongPress,
+        shouldUseHapticsOnPress,
+        nextFocusRef,
+        keyboardShortcut,
+        shouldUseAutoHitSlop,
+        enableInScreenReaderStates,
+        ...rest
+    } = props;
+
+    const isScreenReaderActive = Accessibility.useScreenReaderStatus();
+    const [hitslop, onLayout] = Accessibility.useAutoHitSlop();
+
+    const isDisabled = useMemo(() => {
+        let shouldBeDisabledByScreenReader = false;
+        switch (enableInScreenReaderStates) {
+            case 'all':
+                shouldBeDisabledByScreenReader = false;
+                break;
+            case 'disabled':
+                shouldBeDisabledByScreenReader = !isScreenReaderActive;
+                break;
+            case 'active':
+                shouldBeDisabledByScreenReader = isScreenReaderActive;
+                break;
+            default:
+                break;
+        }
+        return props.disabled || shouldBeDisabledByScreenReader;
+    }, [isScreenReaderActive, enableInScreenReaderStates, props.disabled]);
+
+    const onLongPressHandler = useCallback(() => {
+        if (!onLongPress) {
+            return;
+        }
+        if (shouldUseHapticsOnLongPress) {
+            HapticFeedback.longPress();
+        }
+        onLongPress();
+        Accessibility.moveAccessibilityFocus(nextFocusRef);
+    }, [shouldUseHapticsOnLongPress, onLongPress, nextFocusRef]);
+
+    const onPressHandler = useCallback(() => {
+        if (shouldUseHapticsOnPress) {
+            HapticFeedback.press();
+        }
+        onPress();
+        Accessibility.moveAccessibilityFocus(nextFocusRef);
+    }, [shouldUseHapticsOnPress, onPress, nextFocusRef]);
+
+    const onKeyPressHandler = useCallback((event) => {
+        if (event.key !== 'Enter') {
+            return;
+        }
+        onPressHandler();
+    }, [onPressHandler]);
+
+    useEffect(() => {
+        if (!keyboardShortcut) {
+            return;
```

Comment:
IDK if this matters but maybe we should have a consistent return type here:

```suggestion
            return () => {};
```